draft-ietf-ppm-dap.md

```diff
@@ -1307,6 +1313,10 @@ of the task. The job ID is a 16-byte value, structured as follows:
 opaque AggregationJobID[16];
 ~~~
 
+The set of candidate reports MUST produce a set of report ID ranges which does
```

Comment:
How strict do you think we should be? As we discussed on slack today, at a minimum we want the helper to be able to reject any range that overlaps with an already aggregated range. But the helper could make a smarter decision by "merging" the two ranges and rejecting only the overlap, in which case the MUST here would not be enforced by the helper.

Reply:
I think a `MUST` is appropriate here for the Leader: the Leader must follow this suggestion to avoid having otherwise-acceptable reports rejected as a replay.

The Helper can indeed make a few different choices about how to deal with an aggregation job which does intersect with a previous aggregation job. The relevant text for the Helper is currently:

> Check if the report has been previously aggregated. If so, the input share MUST be marked as invalid with the error `report_replayed`. A report is considered previously aggregated if it intersects with a report ID range for a completed aggregation job (see {{report-id-ranges}}).

... which implies to me that we have to effectively allow "merging" of ranges if they intersect, in the case that an intersecting aggregation job is received. (A Helper which does something more specific, e.g. marking individual report IDs as used, would work to avoid replays; but I'm not sure a Leader could take advantage of this without knowing the specific strategy used by the Helper.)

I'm personally happy with this level of flexibility, but maybe the Helper should be more be more flexible? If so, we might consider relaxing the Leader's MUST here.

Reply:
I prefer MUST here as well.